compiler/rustc_trait_selection/src/solve/eval_ctxt/canonical.rs

```diff
@@ -317,12 +316,8 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
         param_env: ty::ParamEnv<'tcx>,
         opaque_types: &[(ty::OpaqueTypeKey<'tcx>, Ty<'tcx>)],
     ) -> Result<(), NoSolution> {
-        for &(a, b) in opaque_types {
-            let InferOk { value: (), obligations } =
-                self.infcx.register_hidden_type_in_new_solver(a, param_env, b)?;
-            // It's sound to drop these obligations, since the normalizes-to goal
-            // is responsible for proving these obligations.
-            let _ = obligations;
+        for &(key, ty) in opaque_types {
+            self.insert_hidden_type(key, param_env, ty)?;
```

Comment:
this was previously broken. `register_hidden_type` returns both obligations from checking the item bounds and from equating the added hidden type with the previous one. Can't be bothered to write a test for that though :sweat_smile: especially cause wf of opaque types is checked separately anyways ^^

Reply:
oof